lib/t/matches.ex

```diff
@@ -272,6 +303,43 @@ defmodule T.Matches do
     broadcast_from_for_user(by_user_id, {__MODULE__, :unmatched, match_id})
   end
 
+  @spec expire_match(uuid, uuid) :: boolean
+  def expire_match(user_id_1, user_id_2) do
+    m = "match between #{user_id_1} and #{user_id_2} expired"
+    Logger.warn(m)
+    Bot.async_post_message(m)
+
+    Multi.new()
+    |> Multi.run(:unmatch, fn _repo, _changes ->
+      Match
+      |> where(user_id_1: ^user_id_1)
+      |> where(user_id_2: ^user_id_2)
+      |> select([m], %{id: m.id, users: [m.user_id_1, m.user_id_2]})
+      |> Repo.delete_all()
+      |> case do
+        {1, [match]} -> {:ok, match}
+        {0, _} -> {:error, :match_not_found}
+      end
+    end)
+    |> delete_likes()
+    |> insert_expired_match()
+    |> Repo.transaction()
+    |> case do
+      {:ok, %{unmatch: %{id: match_id}}} ->
+        notify_expired([user_id_1, user_id_2], match_id)
+        _expired? = true
+
+      {:error, :unmatch, :match_not_found, _changes} ->
+        _expired? = false
+    end
+  end
+
+  defp notify_expired(users, match_id) do
+    for user_id <- users do
+      broadcast_for_user(user_id, {__MODULE__, :expired, match_id})
+    end
+  end
+
   # TODO cleanup
   defp preload_match_profiles(matches, user_id) do
```

Comment:
probably could be :joined and not preloaded similar to ExpiredMatch

Reply:
not that easy since you gotta know which mate is that (uuid1 or uuid2) - maybe later if need